go/vt/vtgate/planbuilder/testdata/aggr_cases.json

```diff
@@ -6442,5 +6442,119 @@
         "user.user"
       ]
     }
+  },
+  {
+    "QueryType": "SELECT",
+    "Original": "select count(*) from user left join (select col, bar from user_extra limit 10) ue on user.col = ue.col group by user.foo, ue.bar",
+    "Instructions": {
+      "OperatorType": "Aggregate",
+      "Variant": "Ordered",
+      "Aggregates": "sum_count_star(0) AS count(*)",
+      "GroupBy": "(1|2), (3|4)",
+      "ResultColumns": 1,
+      "Inputs": [
+        {
+          "OperatorType": "Projection",
+          "Expressions": [
+            "count(*) * coalesce(count(*), 1) as count(*)",
+            ":0 as foo",
+            ":1 as weight_string(`user`.foo)",
+            ":2 as bar",
+            ":3 as weight_string(ue.bar)"
+          ],
+          "Inputs": [
+            {
+              "OperatorType": "Sort",
+              "Variant": "Memory",
+              "OrderBy": "(0|1) ASC, (2|3) ASC",
+              "Inputs": [
+                {
+                  "OperatorType": "Projection",
+                  "Expressions": [
+                    "`user`.foo as foo",
+                    "weight_string(`user`.foo) as weight_string(`user`.foo)",
+                    "ue.bar as bar",
+                    "weight_string(ue.bar) as weight_string(ue.bar)",
+                    "count(*) as count(*)",
+                    "count(*) as count(*)"
+                  ],
+                  "Inputs": [
+                    {
+                      "OperatorType": "Join",
+                      "Variant": "HashLeftJoin",
+                      "Collation": "binary",
+                      "ComparisonType": "INT16",
+                      "JoinColumnIndexes": "-2,2,3,-1,1",
+                      "Predicate": "`user`.col = ue.col",
+                      "TableName": "`user`_user_extra",
+                      "Inputs": [
+                        {
+                          "OperatorType": "Route",
+                          "Variant": "Scatter",
+                          "Keyspace": {
+                            "Name": "user",
+                            "Sharded": true
+                          },
+                          "FieldQuery": "select count(*), `user`.foo, `user`.col from `user` where 1 != 1 group by `user`.foo",
+                          "Query": "select count(*), `user`.foo, `user`.col from `user` group by `user`.foo",
+                          "Table": "`user`"
+                        },
+                        {
+                          "OperatorType": "Aggregate",
+                          "Variant": "Ordered",
+                          "Aggregates": "count_star(0), any_value(3)",
+                          "GroupBy": "(1|2)",
```

Comment:
I don't think this aggregation is correct. Grouping by `ue.bar` and selecting any value of `ue.col` before the left join feels incorrect. The order of operations should be the left join first, and then the grouping and selecting any value.

Reply:
found it...